src/v2/ion/uiSchemaTransformer.js

```diff
@@ -145,13 +188,27 @@ const createUISchema = (remediationValue = [], factors = [], authenticators = []
     // we get back factorId
     if (ionFormField.name === 'factorId' ||
       ionFormField.name === 'factorProfileId') {
+      const factors = transformedResp.factors && transformedResp.factors.value || [];
       uiSchema.type = 'factorSelect';
       uiSchema.options = createFactorTypeOptions(ionFormField.options, factors);
     }
 
     // similar to `factorId` but `authenticator` is a new way to model factors
     // hence it has different structure
-    if (ionFormField.name === 'authenticator') {
+    if (ionFormField.name === 'authenticator'
```

Comment:
nit: at some point we should have a constants file for these `authenticator` names, `ionFormField ` name and `form` names.

Reply:
Good point!